presto-main/src/main/java/com/facebook/presto/sql/expressions/ExpressionOptimizerManager.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.expressions;
+
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.nodeManager.PluginNodeManager;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.relation.ExpressionOptimizer;
+import com.facebook.presto.spi.relation.ExpressionOptimizerProvider;
+import com.facebook.presto.spi.sql.planner.ExpressionOptimizerContext;
+import com.facebook.presto.spi.sql.planner.ExpressionOptimizerFactory;
+import com.facebook.presto.sql.relational.FunctionResolution;
+import com.facebook.presto.sql.relational.RowExpressionOptimizer;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.facebook.presto.util.PropertiesUtil.loadProperties;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+public class ExpressionOptimizerManager
+        implements ExpressionOptimizerProvider
+{
+    private static final File EXPRESSION_MANAGER_CONFIGURATION = new File("etc/expression-manager.properties");
```

Comment:
Do we really need a separate config file for this? Is there a reason it can't just be rolled into the existing coordinator properties file?

Additionally, I feel like rather than having a flag which explicitly enables this, then having to configure the specific expression optimizer factory is complex. It would be simpler to have a single flag which specifies the `expression-manager-factory.name` and the default for the flag should be the current default behavior. From what I understand it should default to the name for the in-memory expression optimizer.

Reply:
It's a separate config because the implementation will define properties that the Presto runtime can't know in advance.

I think this should be modeled as a plugin because it introduces a degree of separation between Presto and Velox, and also opens the door for third party integrations.